.github/scripts/fetch_and_compare_version.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2019 Intel Corporation.  All rights reserved.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+
+import re
+import shlex
+import subprocess
+import sys
+
+
+def fetch_version_from_code():
+    """
+    search the semantic version definition in build-scripts/config_common.cmake
+    """
+    major, minor, patch = "", "", ""
+    with open("build-scripts/config_common.cmake", encoding="utf-8") as f:
+        for line in f.readlines():
```

Comment:
given we iterate through the whole file (and not just need a single line), this could even be simplified to:
```
for line in f:
```
See [here](https://docs.python.org/3/tutorial/inputoutput.html#methods-of-file-objects) for details.